src/main/java/com/navercorp/arcus/spring/cache/ArcusCache.java

```diff
@@ -117,6 +118,62 @@ public ValueWrapper get(Object key) {
     return (value != null ? new SimpleValueWrapper(value) : null);
   }
 
+  @SuppressWarnings("unchecked")
+  @Override
+  public <T> T get(Object key, Class<T> type) {
+    ValueWrapper valueWrapper = get(key);
+    if (valueWrapper == null) {
+      return null;
+    }
+
+    Object value = valueWrapper.get();
+    if (value == null) {
+      return null;
```

Comment:
> spec을 좀 읽어보니, cache에 NULL 값 저장이 가정되어 있네요.
> 이러한 null 값 저장이 언제부터 spec에 들어간 건지 궁금하네요.
> ARCUS는 현재 null 값 저장 기능이 없는 상태인데,
> 다른 cache는 null 값 저장을 어떻게 다루는 지를 한번 살펴볼 필요가 있네요.
> 
> 여튼, get() 수행에서 valueWrapper를 얻었다면,
> 그 안에 value가 null이더라도 주어진 type에 따라 casting해야 하지 않나요 ?

null이여도 어느 타입으로 주든간에 casting이 가능하네요..

Reply:
Spring 문서를 다시 읽어보니 value 가 null이어도 ValueWrapper는 null로 주지않아야하는 것 같네요. get 메소드에서 ValueWrapper로 무조건 감싸도록 처리하여 동작을 확인해봐야 할 것 같습니다.

Reply:
https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/cache/Cache.html#get-java.lang.Object-
`Returns null if the cache contains no mapping for this key; otherwise, the cached value (which may be null itself) will be returned in a Cache.ValueWrapper.`

이 부분을 말씀하시는거죠? ARCUS에서 null 값 저장이 고려가 되어있지 않기 때문에, null을 ValueWrapper로 감쌀 필요는 없을 것 같네요.

Reply:
그 부분이 맞아요.
ARCUS가 아직 null 값을 지원하지 않는 것은 맞아요.
하지만,
- 해당 key가 존재하지 않으면, valueWrapper가 null로 리턴되고
- 해당 key가 존재하면 valueWrapper가 리턴됩니다.
   - 그 안의 value가 null이라면 (ARCUS는 이 경우가 발생하지 않지만 향후를 위하여 이 코드를 추가한다면), 
   - null 값을 type casting하여 리턴하는 것이 맞다는 것입니다.
   - 향후를 위하지 않는다면, `value == null` 검사하여 null 리턴하는 코드 자체가 필요 없습니다.
      왜냐면, ARCUS에서는 value == null로 하여 valueWrapper를 리턴하지 않기 때문입니다.

Reply:
말씀하신대로 현재로서는 130~131 라인의 코드는 의미가 없어 삭제를 해야 할 것 같네요... 애초에 get 메소드에서 ValueWrapper에 null을 감싸지 않기 때문에 실행 자체가 될 수 없는 코드인 것 같습니다.